src/transformers/modeling_utils.py

```diff
@@ -3975,6 +3976,12 @@ def from_pretrained(
                     _from_pipeline=from_pipeline,
                     **kwargs,
                 )
+                if generation_config is not None:
+                    logger.info(
+                        "Both `pretrained_model_name_or_path` and `generation_config` are provided. The `generation_config`"
+                        " will be used to override the `pretrained_model_name_or_path` generation config."
+                    )
+                    model.generation_config = model.generation_config.from_dict(generation_config.to_dict())
```

Comment:
Request: let's move this a few lines up, to before `if model.can_generate() and pretrained_model_name_or_path is not None:` (converting that `if` to an `elif`)

Reason: we initialize a `generation_config` in `__init__` ([here](https://github.com/huggingface/transformers/blob/b3909989d38cc2666110e7bc6d114d29d942639c/src/transformers/modeling_utils.py#L1402)), and we want to overwrite it with the user-defined generation config when it is passed. In some corner cases, `pretrained_model_name_or_path` can be `None`, and these lines wouldn't be reached as is :)

Reply:
I see. Thanks for the context